### PR TITLE
Limit the markdown !include syntax to only support .md extension

### PR DESCRIPTION
**Pull request by v-caxian**
[AB#336152](https://dev.azure.com/ceapex/Engineering/_workitems/edit/336152)

Limit the markdown !include syntax to only support .md extension, any other file extensions raises a warning.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/docfx/pull/6954)

### CODE REVIEW

File: src/docfx/lib/markdown/IncludeExtension.cs [hunk 1]
```diff
@@ -65,6 +65,12 @@ private static void RemoveIncludeRenderer(IMarkdownRenderer renderer)
                 return;
             }
 
+            if (!inclusionBlock.IncludedFilePath.EndsWith(".md", StringComparison.OrdinalIgnoreCase))
```
Comment (yufeih) [inline src/docfx/lib/markdown/IncludeExtension.cs:68]: Move this check before calling `_context.ReadFile`

File: src/docfx/Errors.cs [hunk 2]
```diff
@@ -505,6 +505,9 @@ public static class Markdown
         {
             public static Error IncludeNotFound(SourceInfo<string?> source)
                 => new Error(ErrorLevel.Error, "include-not-found", $"Invalid include link: '{source}'.", source);
+
+            public static Error IncludeNotSupport(SourceInfo<string?> source)
+                => new Error(ErrorLevel.Warning, "include-not-support", $"Invalid include link extension: '{source}'.", source);
```
Comment (yufeih) [inline src/docfx/Errors.cs:510]:
```suggestion
                => new Error(ErrorLevel.Warning, "include-invalid", $"Invalid include link extension: '{source}'.", source);
```

Comment (v-caxian) [reply src/docfx/Errors.cs]: Moved.

File: src/docfx/lib/markdown/IncludeExtension.cs (hunk 1 above)
Comment (v-caxian) [reply src/docfx/lib/markdown/IncludeExtension.cs]: Moved.

File: docs/specs/resolve.yml [hunk 3]
```diff
@@ -265,6 +265,22 @@ outputs:
...
+  docfx.yml:
+  docs/a.md: |
+    [!include[B](../media/b.png)]
+    Link to [!include[C](../c.ps1)]
+    Empty inclusion [!include[D]()]
+  media/b.png: Token
+  c.ps1: Token
+outputs:
+  .errors.log: |
+    {"message_severity":"warning","code":"include-invalid","message":"Invalid include link extension: '../media/b.png'.","file":"docs/a.md","line":1,"column":1}
```
Comment (v-caxian) [inline docs/specs/resolve.yml:280]: @yufeih Would it be better if the message changed to the following?

```suggestion
    {"message_severity":"warning","code":"include-invalid","message":"Invalid include link extension: '.png', the '!include' syntax only supports '.md' extension.","file":"docs/a.md","line":1,"column":1}
```

**Review by yufeih — APPROVED** [review]

**Pull request merged**